### PR TITLE
Chapter 2 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
* [AP - Chapter 2] Android and Model-View-Controller  - https://androidtraining.atlassian.net/browse/AT-39

## Description:
I've done the chapter 2

## :camera_flash: UI preview
ezgif com-video-to-gif(2)

## :pushpin: Notes:
The code contains the challenges

### CODE REVIEW

File: GeoQuizChallenge/.idea/.name [hunk 1]
```diff
@@ -0,0 +1 @@
+GeoQuiz
```
Comment (FerrariBruno) [inline GeoQuizChallenge/.idea/.name]: this shouldn't be tracked; check your .gitignore

File: GeoQuizChallenge/.idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (FerrariBruno) [inline GeoQuizChallenge/.idea/vcs.xml]: same comment

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml [hunk 3]
```diff
@@ -40,4 +39,26 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        >
+
+        <ImageButton
+            android:id="@+id/prev_button"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+           android:src="@drawable/arrow_left"
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: identation

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml [hunk 4]
```diff
@@ -10,11 +10,10 @@
     >
 
     <TextView
-        android:id="@+id/textView"
+        android:id="@+id/question_text_view"
```
Comment (FerrariBruno) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: As per the style guide, it should be `question_textView`. However, I do think that camelCase would be a better fit, but this is something we need to discuss with the team. @f7deleon @matir91 @agustinaGallo97 
what do you think?

File: GeoQuizChallenge/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,8 +1,15 @@
...
+    <string name="question_asia">Lake Baikal is the world\'s oldest and deepest freshwater lake.</string>
     <string name="question_button_true">true</string>
-    <string name="question_text">Canberra is the capital of Australia.</string>
     <string name="question_button_false">false</string>
+    <string name="prev_button">Prev</string>
+    <string name="next_button">Next</string>
     <string name="correct_toast">Correct!</string>
     <string name="incorrect_toast">Incorrect!</string>
-</resources>
+</resources>
```
Comment (FerrariBruno) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml]: nl

File: GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/Question.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.bignerdranch.android.geoquiz;
```
Comment (mirland) [inline GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/Question.java]: use more packages

File: GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/QuizActivity.java [hunk 7]
```diff
@@ -1,29 +1,72 @@
...
 
 public class QuizActivity extends AppCompatActivity {
   @SuppressWarnings("FieldCanBeLocal")
   private Button trueButton;
   @SuppressWarnings("FieldCanBeLocal")
   private Button falseButton;
+  private ImageButton nextButton;
+  private ImageButton prevButton;
+  private TextView questionTextView;
+  private Question[] questionBank = new Question[] {
```
Comment (mirland) [inline GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/QuizActivity.java]: I'd make a controller to provide this questions, something simple, just a singleton

File: GeoQuizChallenge/app/src/main/res/values/strings.xml (hunk 5 above)
Comment (mirland) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml]: nl

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml (hunk 3 above)
Comment (mirland) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: indentation

File: GeoQuizChallenge/.idea/vcs.xml (hunk 2 above)
Comment (mirland) [inline GeoQuizChallenge/.idea/vcs.xml]: this shouldn't be tracked

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml (hunk 3 above)
Comment (FerrariBruno) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: indentation

File: GeoQuizChallenge/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
+    <string name="question_mideast">The Suez Canal connects the Read Sea and the Indian Ocean</string>
+    <string name="question_africa">The source of the Nile River is in Egypt</string>
+    <string name="question_americas">The Amazon River is the longest river in the Americas.</string>
+    <string name="question_asia">Lake Baikal is the world\'s oldest and deepest freshwater lake.</string>
     <string name="question_button_true">true</string>
-    <string name="question_text">Canberra is the capital of Australia.</string>
     <string name="question_button_false">false</string>
+    <string name="prev_button">Prev</string>
+    <string name="next_button">Next</string>
     <string name="correct_toast">Correct!</string>
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml]: change this to just correct

File: GeoQuizChallenge/app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1,8 +1,15 @@
...
     <string name="incorrect_toast">Incorrect!</string>
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml]: same for this

File: GeoQuizChallenge/app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -1,8 +1,15 @@
...
     <string name="app_name">GeoQuiz</string>
+    <string name="question_australia">Canberra is the capital of Australia.</string>
+    <string name="question_oceans">The Pacific Ocean is larger than the Atlantic Ocean.</string>
+    <string name="question_mideast">The Suez Canal connects the Read Sea and the Indian Ocean</string>
+    <string name="question_africa">The source of the Nile River is in Egypt</string>
+    <string name="question_americas">The Amazon River is the longest river in the Americas.</string>
+    <string name="question_asia">Lake Baikal is the world\'s oldest and deepest freshwater lake.</string>
     <string name="question_button_true">true</string>
-    <string name="question_text">Canberra is the capital of Australia.</string>
     <string name="question_button_false">false</string>
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml]: just false

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml (hunk 4 above)
Comment (mirland) [reply GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: Camel case is fine :)

File: GeoQuizChallenge/app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -1,8 +1,15 @@
...
+    <string name="prev_button">Prev</string>
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/values/strings.xml:11]: as a general comment if the string is a word we ussaly put the same word as the name of the string also i think the style guide recommends you to name the strings with the same text interpolated with `_ ` so `question_africa` should be `the_source_of_the_nile_river_is_in_egypt`

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml [hunk 12]
```diff
@@ -40,4 +39,26 @@
...
+
+        <ImageButton
+            android:id="@+id/prev_button"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+           android:src="@drawable/arrow_left"
+            />
+
+        <ImageButton
+            android:id="@+id/next_button"
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: change to cammel case

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml [hunk 13]
```diff
@@ -40,4 +39,26 @@
...
     </LinearLayout>
 
+    <LinearLayout
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        >
+
+        <ImageButton
+            android:id="@+id/prev_button"
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: same

File: GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml (hunk 4 above)
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/res/layout/activity_quiz.xml]: use cammel

File: GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/QuizActivity.java [hunk 14]
```diff
@@ -1,29 +1,72 @@
 package com.bignerdranch.android.geoquiz;
 
+import androidx.appcompat.app.AppCompatActivity;
+
```
Comment (f7deleon) [inline GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/QuizActivity.java]: rm

**Review by f7deleon — COMMENTED** [review]
Change the pull request name to something more descriptive like added chapter 2 or something like that

File: GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/Controllers/QuestionController.java [hunk 15]
```diff
@@ -0,0 +1,28 @@
+package com.bignerdranch.android.geoquiz.Controllers;
```
Comment (FerrariBruno) [inline GeoQuizChallenge/app/src/main/java/com/bignerdranch/android/geoquiz/Controllers/QuestionController.java:1]:
```suggestion
package com.bignerdranch.android.geoquiz.Controllers;

```

**Review by FerrariBruno — APPROVED** [review]

**Pull request merged**